modules/oe_content_sub_entity/src/Entity/SubEntityBase.php

```diff
@@ -28,7 +30,13 @@ abstract class SubEntityBase extends ContentEntityBase implements SubEntityInter
    */
   public function label() {
     // Return the entity's bundle label by default.
-    return $this->get('type')->entity->label();
+    $label = $this->get('type')->entity->label();
+
+    $event = new SubEntityEvent($this);
+    $event->setLabel($label);
```

Comment:
```suggestion
```
Default label can be generated in the SubEntityEvent because we pass entity there and it will allow to dispatch event in another place without providing default label.